nasa/harmony
# Harmony 1855 - Add STAC catalogs when status is complete_with_errors

## Jira Issue ID
HARMONY-1855

## Description
Fixes a bug where jobs completing with errors did not have a STAC catalog link in the job results page

## Local Test Steps
1. Run the following query
```
http://localhost:3000/C1234724470-POCLOUD/ogc-api-coverages/1.0.0/collections/all/coverage/rangeset?subset=lat(-45%3A45)&subset=lon(0%3A180)&forceAsync=true&granuleId=G1236546414-POCLOUD&granuleId=G1236614412-POCLOUD
```
2. Go to the job status page once the job completes and make sure it has a STAC catalog link and that the STAC catalog has one item in it (of the two granules, one failed).

## PR Acceptance Checklist
